File: frontend/src/api/police.ts
import { apiPost } from './request'

export interface PoliceEventQuery {
  start_time?: string
  end_time?: string
  keyword?: string
  event_type?: string
  unit_code?: string
  page: number
  page_size: number
}

export function searchPoliceEvents(query: PoliceEventQuery) {
  return apiPost<{ total: number; page: number; page_size: number; items: unknown[] }>('/police-events/search', query)
}

export function getPoliceOverview(query: PoliceEventQuery) {
  return apiPost<{ total: number; by_type: unknown[]; by_unit: unknown[]; trend: unknown[] }>('/police-events/overview', query)
}

File: backend/app/schemas/police.py
from datetime import datetime

from pydantic import BaseModel, Field


class PoliceEventQuery(BaseModel):
    start_time: datetime | None = None
    end_time: datetime | None = None
    keyword: str | None = None
    event_type: str | None = None
    unit_code: str | None = None
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=200)


class PoliceEventItem(BaseModel):
    id: str
    alarm_time: datetime | None = None
    event_type: str | None = None
    unit_name: str | None = None
    address: str | None = None
    summary: str | None = None


class PoliceOverview(BaseModel):
    total: int
    by_type: list[dict] = Field(default_factory=list)
    by_unit: list[dict] = Field(default_factory=list)
    trend: list[dict] = Field(default_factory=list)
